config/routes.rb

```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
   get 'companies/:company_id/shipments', to: 'shipments#index'
   get 'companies/:company_id/shipments/:shipment_id', to: 'shipments#show'
+  get 'companies/:company_id/trace/:shipment_id', to: 'shipments#trace'
```

Comment:
Maybe this is more explicit?
```suggestion
  get 'companies/:company_id/shipments/:id/trace', to: 'shipments#trace'
```

Reply:
Your suggest is great! I'll modify this part